test/Compiler/Tests.hs

```diff
@@ -62,9 +67,9 @@ data TestOptions
     } deriving (Show, Read)
 
 allCompilers :: [Compiler]
-allCompilers =
-  map MAlonzo [Lazy, StrictData, Strict] ++
-  map JS [NonOptimized, Optimized, MinifiedOptimized]
+allCompilers
+  =  [ MAlonzo strict | strict <- [minBound..]]
```

Comment:
This changes the order from Lazy, StrictData, Strict to Strict | StrictData | Lazy.
It could be restored by reordering the constructors of `Strict`, or reverting this line.
I'll opt for the latter:  It is often better to be more explicit in the code for the sake of the reader.

Reply:
Btw, nested Enums are now supported with `DerivingVia`, see e.g. https://github.com/agda/agda/blob/4b1da6a4d032bfa40f07a76af31c999614246383/src/full/Agda/Interaction/Options/Errors.hs#L286

Reply:
@lawcho: You marked this conversation as resolved.
Resolving a conversation should be preceded by a resolution of the conversation, so either a code change or an agreement that none is necessary.
I wrote:
> This changes the order from Lazy, StrictData, Strict to Strict | StrictData | Lazy.
> It could be restored by reordering the constructors of Strict, or reverting this line.
> I'll opt for the latter: It is often better to be more explicit in the code for the sake of the reader.

Reply:
I agree with making the `[Lazy, StrictData, Strict]` order explicit.
Apologies for resolving too early -- I read your comment as "I plan to revert this line myself, do you agree"?
I'll push the change to this branch.